src/markets.rs

```diff
@@ -0,0 +1,150 @@
+mod kraken;
+use chrono::{DateTime, Utc};
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, strum_macros::Display)]
+pub enum TradingPair {
+    BtcDai,
+}
+
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub struct Rate {
+    trading_pair: TradingPair,
+    position: Position,
+    rate: f64,
+    timestamp: DateTime<Utc>,
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, strum_macros::Display)]
+pub enum Position {
+    BUY,
+    SELL,
+}
+
+pub struct Ohlc {
+    high: f64,
+    low: f64,
+    vwap: f64,
+    timestamp: DateTime<Utc>,
+    trading_pair: TradingPair,
+}
+
+impl Ohlc {
+    fn to_rate(&self, position: Position) -> anyhow::Result<Rate> {
+        let rate = if self.vwap == 0.0 {
+            let precision = 0.000_000_000_000_001;
+            if (self.high - self.low).abs() > precision {
+                tracing::warn!("OHLC high and low value are not the same even though there were no trades recorded (vwap 0).")
+            }
+            self.high
+        } else {
+            self.vwap
+        };
+
+        let trading_pair = self.trading_pair;
+        let timestamp = self.timestamp;
+
+        match position {
+            Position::BUY => Ok(Rate {
+                trading_pair,
+                position,
+                rate: 1f64 / rate,
```

Comment:
```suggestion
                rate: rate.recip(),
```
Doing float maths here! Not sure it's what we want.

Reply:
Maybe doing floating point arithmetic here is fine because we're just obtaining a "good" rate to use for our orders. It's not like we're losing our own money by being imprecise. Does that make any sense?

Reply:
I left this one as is for now